src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -968,12 +990,22 @@ private void WriteHeaders(HttpRequestMessage request)
 
         private async ValueTask<Http2Stream> SendHeadersAsync(HttpRequestMessage request, CancellationToken cancellationToken, bool mustFlush = false)
         {
-            // Ensure we don't exceed the max concurrent streams setting.
-            await _concurrentStreams.RequestCreditAsync(1, cancellationToken).ConfigureAwait(false);
-
-            // We serialize usage of the header encoder and the header buffer separately from the
-            // write lock
-            await _headerSerializationLock.WaitAsync(cancellationToken).ConfigureAwait(false);
+            try
+            {
+                // Ensure we don't exceed the max concurrent streams setting.
+                await _concurrentStreams.RequestCreditAsync(1, cancellationToken).ConfigureAwait(false);
+                // We serialize usage of the header encoder and the header buffer separately from the
+                // write lock
+                await _headerSerializationLock.WaitAsync(cancellationToken).ConfigureAwait(false);
+            }
+            catch (ObjectDisposedException)
```

Comment:
We can't avoid throwing this in the first place?

Reply:
I think we can. I would suggest to do that as part #38267.

Reply:
If we don't think this is the right change, then why are we doing anything here right now? Why is it necessary to make a temporary change we will back out later?

Reply:
we have randomly failing tests and that impacts other developers. We also seen this with gRPC and we want to stabilize for Monday preview cutoff